apps/condo/domains/ticket/tasks/exportIncidents.js

```diff
@@ -0,0 +1,234 @@
+const DATE_FORMAT = 'DD.MM.YYYY'
+const EMPTY_VALUE = '—'
+
+const taskLogger = getLogger('exportIncidents')
+
+
+const buildTranslations = (locale) => ({
+    AllProperties: i18n('incident.fields.properties.allSelected', { locale }),
+    Actual: i18n('incident.status.actual', { locale }),
+    NotActual: i18n('incident.status.notActual', { locale }),
+    WorkTypes: {
+        [INCIDENT_WORK_TYPE_SCHEDULED]: i18n('incident.workType.scheduled', { locale }),
+        [INCIDENT_WORK_TYPE_EMERGENCY]: i18n('incident.workType.emergency', { locale }),
+    },
+})
+
+const classifiersToString = (classifiers = []) => {
+    if (classifiers.length < 1) {
+        return EMPTY_VALUE
+    }
+
+    const categories = uniq(classifiers
+        .map(({ category }) => category))
+        .join(', ')
+    const problems = uniq(classifiers
+        .map(({ problem }) => problem))
+        .join(', ')
+
+    const renderProblems = problems ? ` » ${problems}` : ''
+
+    return `${categories}${renderProblems}`
+}
+
+const formatDate = (date, timeZone, format = DATE_FORMAT) => {
+    return dayjs(date).tz(timeZone).format(format)
+}
+
+const incidentToRow = async ({ task, incident, translations }) => {
+    const { timeZone } = task
+
+    const incidentProperties = await loadIncidentPropertiesForExcelExport({
+        where: {
+            incident: { id: incident.id },
+            deletedAt: null,
+        },
+    })
+
+    const incidentClassifierIncidents = await loadIncidentClassifierIncidentsForExcelExport({
+        where: {
+            incident: { id: incident.id },
+            deletedAt: null,
+        },
+    })
+
+    const classifierIds = uniq(incidentClassifierIncidents.map(({ classifier }) => classifier))
+
+    const classifiers = await loadIncidentClassifiersForExcelExport({
+        where: {
+            id_in: classifierIds,
+        },
+    })
+
+    const properties = incidentProperties
+        .map(({ property }) => property)
+    const isActual = incident.status === INCIDENT_STATUS_ACTUAL
+
+    return {
+        number: incident.number,
+        addresses: incident.hasAllProperties
+            ? translations.AllProperties
+            : properties.join(';\n'),
+        classifiers: classifiersToString(classifiers),
+        details: incident.details,
+        status: isActual ? translations.Actual : translations.NotActual,
+        workStart: incident.workStart ? formatDate(incident.workStart, timeZone) : EMPTY_VALUE,
+        workFinish: incident.workFinish ? formatDate(incident.workFinish, timeZone) : EMPTY_VALUE,
+        workType: incident.workType ? translations.WorkTypes[incident.workType] : EMPTY_VALUE,
+        organization: incident.organization,
+        textForResident: incident.textForResident || EMPTY_VALUE,
+        createdBy: incident.createdBy,
+        createdAt: incident.createdAt ? formatDate(incident.createdAt, timeZone) : EMPTY_VALUE,
+    }
+}
+
+const buildExportFile = async ({ task, rows }) => {
```

Comment:
Same as `useIncidentExportTaskUIInterface`.
Should we do something common and not copy the code?

Reply:
I think in this case it's best to leave it as it is.

Reply:
There are more places that can change, unlike `useIncidentExportTaskUIInterface `